pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go

```diff
@@ -59,39 +59,27 @@ func (c *clusterRefreshController) onClusterChange(key string, cluster *mgmtv3.C
 		// not a KEv2 cluster
 		return cluster, nil
 	}
-
 	if !ready {
-		return cluster, nil
-	}
-
-	var lastRefreshTime string
-	var err error
-
-	if cluster.Annotations != nil {
-		lastRefreshTime = cluster.Annotations[clusterLastRefreshTime]
-	}
-
-	providerRefreshInterval, err := getProviderRefreshInterval(provider)
-	if err != nil {
-		return cluster, err
+		return c.updateCluster(cluster)
 	}
 
-	shouldRefresh, err := shouldRefreshCluster(providerRefreshInterval, lastRefreshTime)
+	providerRefreshInterval := getProviderRefreshInterval(provider)
+	nextRefresh, err := nextRefreshTime(providerRefreshInterval, cluster.Annotations[clusterLastRefreshTime])
 	if err != nil {
 		return cluster, err
 	}
 
-	if shouldRefresh {
-		return cluster, err
-	}
-
-	cluster, err = c.refreshClusterUpstreamSpec(cluster, provider)
-	if err != nil {
-		return cluster, err
+	now := time.Now()
+	if nextRefresh.Before(now) {
+		cluster, err = c.refreshClusterUpstreamSpec(cluster, provider)
+		if err != nil {
+			return cluster, err
+		}
+		c.clusterEnqueueAfter(key, providerRefreshInterval)
+	} else {
+		c.clusterEnqueueAfter(key, nextRefresh.Sub(now))
```

Comment:
> This fixes an issue where the refreshing would stop if a cluster is updated successfully before a refresh is needed (the cluster would enter the handler because the cluster object was updated, the refresh time was not exceed so the handler would return without re-queuing). 

I take it thats what is going on here. Really its the removal of line 84.

Reply:
Yes, you have this correct.